cosmos/dbt_adapters/__init__.py

```diff
@@ -0,0 +1,18 @@
+from __future__ import annotations
+
+from typing import Any
+
+from cosmos.constants import BIGQUERY_PROFILE_TYPE
+from cosmos.dbt_adapters.bigquery import _associate_bigquery_async_op_args, _mock_bigquery_adapter
+
+PROFILE_TYPE_MOCK_ADAPTER_CALLABLE_MAP = {
```

Comment:
We should not do this type of mapping again.

I had given @pankajastro similar feedback on https://github.com/astronomer/astronomer-cosmos/pull/1474/files#r1941988073

He implemented the "automatic mapping":
https://github.com/astronomer/astronomer-cosmos/blob/c344eb4750f0269ecd267326373814abf56ec18e/cosmos/operators/_asynchronous/base.py#L16-L36

By doing https://github.com/astronomer/astronomer-cosmos/pull/1474/files#r1941988073 my hope is that this is removed.

Reply:
thanks @tatiana . I have created a follow-up task to work on immediately on this https://github.com/astronomer/astronomer-cosmos/issues/1508